common/konfig.py

```diff
@@ -0,0 +1,447 @@
+        name_items = filter(
+            lambda val:
+                val[0].startswith('profile') and val[0].endswith('name'),
+            self._conf.items()
+        )
+        self._profiles = {
+            name: int(pid.replace('profile', '').replace('.name', ''))
+            for pid, name in name_items
+        }
+
+    def __getitem__(self, key: str) -> Any:
+        try:
+            return self._conf[key]
+        except KeyError:
+            return self._DEFAULT_VALUES[key]
+
+    def __setitem__(self, key: str, val: Any) -> None:
+        self._conf[key] = val
+
+    def profile(self, name_or_id: Union[str, int]) -> Profile:
+        if isinstance(name_or_id, int):
+            profile_id = name_or_id
+        else:
+            profile_id = self._profiles[name_or_id]
+
+        return Profile(profile_id=profile_id, config=self)
+
+    @property
+    def profile_names(self) -> list[str]:
+        return list(self._profiles.keys())
+
+    @property
+    def profile_ids(self) -> list[int]:
+        return list(self._profiles.values())
+
+    def load(self):
+        """Load configuration from file like object."""
+        @contextlib.contextmanager
+        def _path_or_buffer(path_or_buffer: Union[Path, StringIO]
+                            ) -> Union[TextIOWrapper, StringIO]:
+            """Using a path or a in-memory file (buffer) with a with
+            statement."""
+            try:
+                # It is a regular file
+                path_or_buffer = path_or_buffer.open('r', encoding='utf-8')
+                print(f'{type(path_or_buffer)=}')
+
+            except AttributeError:
+                # Assuming a StringIO instance as in-memory file
+                pass
+
+            yield path_or_buffer
+
+            try:
+                # regular file: close it
+                path_or_buffer.close()
+
+            except AttributeError:
+                # in-memory file: "cursor" back to first byte
+                path_or_buffer.seek(0)
+
+        self._config_parser = configparser.ConfigParser(
+            defaults={'profile1.name': _('Main profile')})
+
+        with _path_or_buffer(self._path) as handle:
+            print(handle)
+            content = handle.read()
+            logger.debug(f'Configuration read from "{self._path}".')
+
+        # Add section header to make it a real INI file
+        self._config_parser.read_string(f'{self._DEFAULT_SECTION}\n{content}')
+
+        # The one and only main section
+        self._conf = self._config_parser['bit']
+
+    def save(self):
+        """Store configuraton to the config file."""
+
+        raise NotImplementedError('Prevent overwritting real config data.')
+
+        buffer = StringIO()
+        self._config_parser.write(buffer)
+        buffer.seek(0)
+
+        with self._path.open('w', encoding='utf-8') as handle:
+            # Write to file without section header
+            handle.write(''.join(buffer.readlines()[1:]))
```

Comment:
Could you just throw away the unwanted first line by advancing the stream position with `readline`? Then you can just `read()` the rest of the buffer directly, without needing to convert it to a list, then join it back together again.

```suggestion
            # Discard unwanted first line
            buffer.readline()
            handle.write(buffer.read())
```